### PR TITLE
SouthAfricaclass1/AashiqAbrahams Js-core1-week2

### PR DESCRIPTION
**Pull request by aashiqabrahams123**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:Aashiq Abrahams
- Your City:South Africa
- Your Slack Name:Aashiq Abrahams

# Homework Details

- Module:Js-core1
- Week:2

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: exercises/C-comparison-operators/exercise.js [hunk 1]
```diff
@@ -7,14 +7,14 @@
 
 var studentCount = 16;
 var mentorCount = 9;
-var moreStudentsThanMentors; // finish this statement
+var moreStudentsThanMentors = studentCount > mentorCount; // finish this statement
 
 var roomMaxCapacity = 25;
-var enoughSpaceInRoom; // finish this statement
+var enoughSpaceInRoom = roomMaxCapacity > enoughSpaceInRoom; // finish this statement
```
Comment (RyanS-Cpt) [inline exercises/C-comparison-operators/exercise.js:13]: You can say greater than and equal to here Aashiq = '>=' as the capacity does not have to bigger than the amount to be true.

File: exercises/I-array-properties/exercise.js [hunk 2]
```diff
@@ -6,7 +6,7 @@
 */
 
 function isEmpty(arr) {
-  return; // complete this statement
+  return arr.length === 0; // complete this statement
 }
```
Comment (RyanS-Cpt) [inline exercises/I-array-properties/exercise.js:10]: This works, but I would suggest rather using greater than 0 Aashiq.

File: mandatory/1-fix-functions.js [hunk 3]
```diff
@@ -10,8 +10,8 @@
 
 */
 
-function mood() {
-  let isHappy = true;
+function mood(moodInput) {
+  let isHappy = moodInput;
 
   if (isHappy) {
```
Comment (RyanS-Cpt) [inline mandatory/1-fix-functions.js:16]: I'm not sure if this code will be very accurate Aashiq

File: mandatory/1-fix-functions.js [hunk 4]
```diff
@@ -31,20 +31,20 @@ function greaterThan10(num) {
...
-  let sliced;
+  let sliced = numbers.sliced(0.5);
 
   return sliced;
 }
 
 function get3rdIndex(arr) {
   let index = 3;
-  let element;
+  let element = arr[index];
```
Comment (RyanS-Cpt) [inline mandatory/1-fix-functions.js:48]: I think you can reduce this function to one line if we know that we want the third index everytime.

File: mandatory/2-function-creation.js [hunk 5]
```diff
@@ -5,7 +5,13 @@ Write a function that:
 - removes any forward slashes (/) in the strings
 - makes the string all lowercase
 */
-function tidyUpString(strArr) {}
+function tidyUpString(strArr) {
+	let result = strArr;
+	for (var i=0; i<result.length; i++) {
+		result[i] = result[i].trim().replace(/\//g, '').toLowerCase();
```
Comment (RyanS-Cpt) [inline mandatory/2-function-creation.js:11]: Very nice chaining methods Aashiq

**Comment by dschouw** [comment]
# Javascript Core 1 - Week 2 Feedback

Well done on completing the homework Aashiq :thumbsup:

I've added some comments below and marked this assignment as Returned on Google Classroom.
Please feel free to reach out to Chandre or me via Slack or Email if anything is unclear.

## Git Status

Total Commits : 10

## Exercises Comments :memo:

1. **Issue** exercises/G-conditionals-2
   exercise-2.js : The code fails on test case where the value/grade is equal to 50. The comparison operator on line 11 should include an equality, i.e. less-than-or-equal-to comparison.

## Mandatory Homework Comments

1. **Issue** 1-fix-function.js
   Function _first5_ attempts to use the _sliced_ function , which causes an error. The code is probably intending to use the [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) function.

## Links :bookmark:

1. [Lesson](https://syllabus.codeyourfuture.io/js-core-1/week-2/lesson)
2. [Code Your Future Coding Style guidelines](https://syllabus.codeyourfuture.io/guides/code-style-guide)

**Pull request closed without merging**